src/daft-dsl/src/functions/function_args.rs

```diff
@@ -388,11 +391,110 @@ impl<T> FunctionArgs<T> {
     }
 }
 
+impl FunctionArgs<Series> {
+    /// Uses serde to extract a scalar value from the function args.
+    /// This will error if the series is not scalar. (len ==1) or if it is not deserializable to the provided type.
+    /// It will also return an error if the value does not exist.
+    pub fn extract<V: FromLiteral, Key: FunctionArgKey>(&self, position: Key) -> DaftResult<V> {
+        let value = position.required(self).map_err(|_| {
+            DaftError::ValueError(format!(
+                "Expected a value for the required argument at position `{position:?}`"
+            ))
+        })?;
+
+        let lit = LiteralValue::try_from_single_value_series(value)?;
+
+        let res = V::try_from_literal(&lit);
+        res.map_err(|e| e.into())
+    }
+
+    /// Uses serde to extract an optional scalar value from the function args.
+    /// This will error if the series is not scalar. (len ==1) or if it is not deserializable to the provided type.
+    /// if the value does not exist, None is returned.
+    pub fn extract_optional<V: FromLiteral, Key: FunctionArgKey>(
+        &self,
+        position: Key,
+    ) -> DaftResult<Option<V>> {
+        let value = position.optional(self).map_err(|_| {
+            DaftError::ValueError(format!(
+                "Expected a value for the optional argument at position `{position:?}`"
+            ))
+        })?;
+
+        match value {
+            Some(value) => {
+                let lit = LiteralValue::try_from_single_value_series(value)?;
+                let res = V::try_from_literal(&lit);
+                res.map_err(|e| e.into()).map(Some)
+            }
+            None => Ok(None),
+        }
+    }
```

Comment:
I see that `try_from_literal` currently only allows conversions from the exact type, and we potentially want to allow for conversions from other types as well. Is this something we plan on changing? Perhaps we also want value-to-literal and literal-to-value to be implemented in separate traits. For example, `usize` to literal doesn't necessarily make sense and probably won't add much value, but literal to `usize` is pretty clear to me and could be useful.

Reply:
I could see that working but I'm not sure if we want to use serde in this way. Having custom serde logic brings a lot of complexity already, and tacking conversion logic onto that would make it even harder to reason through. I think I would rather have the deserialization separate from conversion.

Reply:
also, this doesn't limit us from using the expr directly. you can still do something like

```rs

let arg: ExprRef = inputs.required(0);
// manually parse the arg for whatever reason.

```